selftests/functional/test_plugin_assets.py

```diff
@@ -89,6 +90,28 @@ def test_asset_fetch_success(self):
         self.assertEqual(expected_rc, result.exit_status)
         self.assertIn(expected_output, result.stdout_text)
 
+    def test_asset_register_by_name_fail(self):
+        """Test register command failure."""
+        url = "https://urlnotfound"
+        random = str(uuid.uuid4())
+        cmd_line = "%s assets register %s %s" % (AVOCADO, random, url)
+        result = process.run(cmd_line)
+
+        self.assertIn("Failed to fetch",
+                      result.stderr_text)
+
+    @unittest.skipIf(not os.path.isfile('/etc/hosts'),
+                     " Missing /etc/hosts")
+    def test_asset_register_by_name_success(self):
+        """Test register command success."""
+        url = "/etc/hosts"
+        random = str(uuid.uuid4())
+        cmd_line = "%s assets register %s %s" % (AVOCADO, random, url)
```

Comment:
This is permanently leaving a `/etc/hosts` registered in the cache.  Maybe set a configuration file pointing the cache to the test's temporary directory here?